### PR TITLE
sheldon: 0.7.3 -> 0.7.4

### PR DESCRIPTION
**Pull request by alezkv**
## Description of changes

Build on Unix instead of solely focusing on Linux.

Release notes:

- [Support optional access in templates.][4038307] This allows you to use the
  `.?` operator in templates to optionally access a field in templates, for
  example `{{ hooks?.pre }}` will return `None` and not error if `pre` is not
  present in `hooks`. You will now receive a deprecation warning if you use the
  `get` filter which does the same thing.

  Custom templates might have to be updated, for example if you were using a
  custom `defer` template you would need to update it to the following.

  ```toml
  [templates]
  defer = "{{ hooks?.pre | nl }}{% for file in files %}zsh-defer source \"{{ file }}\"\n{% endfor %}{{ hooks?.post | nl }}"
  ```

[4038307]: https://github.com/rossmacarthur/sheldon/commit/40383074c54c3935908bf76401b42878285510cd

Release diff:

https://github.com/rossmacarthur/sheldon/compare/0.7.3...0.7.4

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by seqizz — APPROVED** [review]

File: pkgs/tools/misc/sheldon/default.nix [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
   pname = "sheldon";
-  version = "0.7.3";
+  version = "0.7.4";
 
   src = fetchFromGitHub {
     owner = "rossmacarthur";
     repo = pname;
-    rev = version;
-    hash = "sha256-vGFR8NL3bOCUuNr0KQuAbjQMxvFbN/T9aVmf7Wxt9JU=";
+    rev = "9a56dcfc79c6e9f9bcf8480c162c50db013bb172";
```
Comment (sikmir) [inline pkgs/tools/misc/sheldon/default.nix:19]:
```suggestion
    rev = version;
```

Comment (alezkv) [reply pkgs/tools/misc/sheldon/default.nix:19]: @sikmir @seqizz I thought that having "more stable" variable like a hash here is better. Do I miss smt here?

Comment (sikmir) [reply pkgs/tools/misc/sheldon/default.nix:19]: What do you mean "more stable"? git tag is a such one.

Comment (alezkv) [reply pkgs/tools/misc/sheldon/default.nix:19]: never mind, hash will cover this case

**Review by sikmir — APPROVED** [review]

**Review by sikmir — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 287249` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sheldon</li>
  </ul>
</details>

```
       > failures:
       >     deprecated_directories
       >     lock_and_source_deprecated_get_filter
       >
       > test result: FAILED. 7 passed; 2 failed; 0 ignored; 0 measured; 12 filtered out; finished in 0.02s
```

Result of `nixpkgs-review pr 287249` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sheldon</li>
  </ul>
</details>

```
       > failures:
       >     deprecated_directories
       >
       > test result: FAILED. 8 passed; 1 failed; 0 ignored; 0 measured; 12 filtered out; finished in 2.72s
```

**Comment by Aleksanaa** [comment]
@ofBorg build sheldon

**Comment by wegank** [comment]
Superseded by #337869.

**Pull request closed without merging**